packages/alea-frontend/pages/search/[courseId].tsx

```diff
@@ -0,0 +1,235 @@
+import SearchIcon from '@mui/icons-material/Search';
+import {
+  Box,
+  IconButton,
+  InputAdornment,
+  LinearProgress,
+  TextField,
+  Typography,
+} from '@mui/material';
+import Tooltip from '@mui/material/Tooltip';
+import {
+  getCourseInfo,
+  getDocumentSections,
+  getRagResponse,
+  SectionsAPIData,
+} from '@stex-react/api';
+import {
+  DocumentWidthSetter,
+  ExpandableContent,
+  mmtHTMLToReact,
+  ServerLinksContext,
+} from '@stex-react/stex-react-renderer';
+import { PRIMARY_COL, XhtmlContentUrl } from '@stex-react/utils';
+import { NextPage } from 'next';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import { useCallback, useContext, useEffect, useState } from 'react';
+import MainLayout from '../../layouts/MainLayout';
+
+interface Reference {
+  archive: string;
+  filepath: string;
+  courseId: string;
+}
+
+function ResultDocument(
```

Comment:
Why not make this a component? Returning jsx from regular functions causes lifecycle issues